python/GUARDDUTY_UNTREATED_FINDINGS/GUARDDUTY_UNTREATED_FINDINGS.py

```diff
@@ -0,0 +1,509 @@
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty Low Severity finding is older than {} days'.format(valid_rule_parameters['daysLowSev'])))
+            continue
+
+        # 4.0 to 6.9 severity score is MediumSev
+        if 4.0 <= each_finding['Severity'] <= 6.9:
+            if days_since_created <= valid_rule_parameters['daysMediumSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty Medium Severity finding is older than {} days'.format(valid_rule_parameters['daysMediumSev'])))
+            continue
+
+        # 7.0 to 8.9 severity score is HighSev
+        if 7.0 <= each_finding['Severity'] <= 8.9:
+            if days_since_created <= valid_rule_parameters['daysHighSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty High Severity finding is older than {} days'.format(valid_rule_parameters['daysHighSev'])))
+            continue
+
+    return evaluations
+
+
+# get number of days since the finding is created
+def get_delta_days(finding_creation_time):
+    current_datetime_object = datetime.utcnow()
+    creation_datetime_object = datetime.strptime(finding_creation_time, '%Y-%m-%dT%H:%M:%S.%fZ')
+    delta_time = current_datetime_object - creation_datetime_object
+    return delta_time.days
+
+# function to get all findings. Each call supports maximum of 50 items.
+def list_all_findings(guardduty_client, guardduty_detector_id):
+    all_findings_list = []
+    findings_id_list = guardduty_client.list_findings(DetectorId=guardduty_detector_id)
+    if not findings_id_list['FindingIds']:
+        return []
+
+    findings_list = guardduty_client.get_findings(DetectorId=guardduty_detector_id, FindingIds=findings_id_list['FindingIds'])
+    all_findings_list += findings_list['Findings']
+    while True:
+        if findings_id_list['NextToken']:
+            findings_id_list = guardduty_client.list_findings(DetectorId=guardduty_detector_id, NextToken=findings_id_list['NextToken'])
+            findings_list = guardduty_client.get_findings(DetectorId=guardduty_detector_id, FindingIds=findings_id_list['FindingIds'])
+            all_findings_list += findings_list['Findings']
+        else:
+            break
+
+    return all_findings_list
+
+
+def evaluate_parameters(rule_parameters):
+    default_days_low_sev = 30
+    default_days_medium_sev = 7
+    default_days_high_sev = 1
+
+    if "daysLowSev" not in rule_parameters:
+        rule_parameters['daysLowSev'] = default_days_low_sev
+    elif not validate_parameter_value(rule_parameters['daysLowSev']):

```

Comment:
I think this should be else or if not. I think pylint would throw unnecessary elif.

Reply:
I dont think we can use `else` here as we need to validate the parameter and make the decision. Let me know if missed something here.